lang/cs/Org.Apache.REEF.IO.TestClient/HadoopFileInputPartitionTest.cs

```diff
@@ -55,6 +55,7 @@ internal static bool TestWithByteDeserializer()
                     .Set(FileSystemInputPartitionConfiguration<IEnumerable<byte>>.FilePathForPartitions, remoteFilePath1)
                     .Set(FileSystemInputPartitionConfiguration<IEnumerable<byte>>.FilePathForPartitions, remoteFilePath2)
                     .Set(FileSystemInputPartitionConfiguration<IEnumerable<byte>>.FileSerializerConfig, serializerConfString)
+                    .Set(FileSystemInputPartitionConfiguration<IEnumerable<byte>>.TempFileFolder, @".\reef\mytest\")
```

Comment:
> Any reason that approach is preferable over what's done above by Julia?

Several:

 * Having an interface like I suggested on the JIRA allows for dynamic behavior, e.g. choosing a temp folder at random from a set of provided choices. Some clusters have many data drives, which makes this desirable.
* This exposes the exact path to the app. This can foster bad coding practices where apps assume to know where temp files will always be. That can be avoided if the temp file creation is the API, not its location.
* As I said on the JIRA, this only attacks the issue for one particular instance of the problem. What if there are other parts of REEF or the application that need temp files? We will see repeats of this very same issue.